### ISSUE TITLE
Training-error computation in NN

### ISSUE THREAD

**Issue by trunda-o** [jdermody/brightwire#17]
Hi.
I've encountered an issue with computing training-error during neural network training. 
The thing is, training-error is computed by a different formula than test-error. The test-error is simply an average of error over the whole test set (file "TrainingEngine.cs", lines 185-187), while training-error computes average of error over only those elements in training set that have non-zero error (file "TrainingEngine.cs", lines 115-122).
This causes the training error to be greater than test-error which seems counter-intuitive to me. Furthermore, this way the training-error always stays high, no matter how many samples in the training set are correctly answered.
I'm using both training and test-error to adjust learning rate and detect overfitting, so I've "fixed" this in my copy, but I wonder if this behavior was intended or not. Does it improve performance of some learning algorithms, or is there some other reason?
(I'm using standard feed-forward network for regression task, with MSE error measure.)

Thanks :)

**Comment by jdermody** [comment]
Hi

Thanks for raising that. It looks like the training error is wrong given the check for zero, which I don't believe should be there. I'll get this fixed!

**Comment by jdermody** [comment]
Fixed in 2.0.6
